src/main/scala/com/adobe/api/platform/runtime/mesos/MesosClient.scala

```diff
@@ -456,11 +494,23 @@ trait MesosClientActor extends Actor with ActorLogging with MesosClientConnectio
 
       }
 
+      //store a reference of last memory offer (total) from each agent
+      val newOfferStats = MesosClient.getOfferStats(config, role, agentOfferMap)
+      //log the agents that previously had offers, but no longer have offers, and are not yet pruned
+      val diff = newOfferStats.keySet.diff(agentOfferMap.keySet)
+      if (diff.nonEmpty) {
+        log.info(s"some agents not included in offer cycle: ${diff} ")
+      }
+      //update agentOfferHistory (replace existing agents' data, add new agents' data)
+      agentOfferHistory = agentOfferHistory ++ newOfferStats
```

Comment:
what would the program miss if it would discard the history and use instead  ```agentOfferHistory = newOfferStats``` ?

Reply:
Is the reason for keeping the history caused by Mesos sending _partial_ offers from a subset of agents ?

Reply:
"subset of agents" is the key, there is no such thing as "partial" offer, I think. Also, there may be cases where there are offers that are not "usable" meaning either they don't have all of mem/cpus/ports resources, or they are not allocated to our specific role, and instead are available to the default * role.

Reply:
now I get it. It would be nice to update the comment to specify this, so that another reader doesn't get confused of this Mesos behavior.